src/common/lib/keymaps.c

```
@@ -23,95 +23,190 @@
  * @return The key code found in the keymap.
  */
-static uint8_t keymap_search_layers(uint8_t row, uint8_t col) {
-  uint8_t key_code = keymap_map[keymap_layer][row][col];
-
-  if (keymap_layer > 0 && key_code == KC_TRNS) {
-    uint8_t layer_key_code;
-    for (int i = keymap_layer; i >= 0; i--) {
-      layer_key_code = keymap_map[i][row][col];
-      if (layer_key_code != KC_TRNS) {
-        key_code = layer_key_code;
+static uint8_t keymap_search_layers(uint8_t row, uint8_t col, uint8_t *source_layer) {
```

Clang (14.0.6)

[warning] function 'keymap_search_layers' has cognitive complexity of 36 (threshold 25)

(readability-function-cognitive-complexity)